bin/kick_off.sh

```diff
@@ -119,6 +123,44 @@ do
 		mkdir tf
 		echo ansible-playbook -i ./inventory --extra-vars "working_dir=${curdir} ansible_python_interpreter=/usr/bin/python3" ten_of_us.yml
 		ansible-playbook -i ./inventory --extra-vars "working_dir=${curdir} ansible_python_interpreter=/usr/bin/python3 delete_tf=none" ten_of_us.yml
+		#
+		if [ $spot_recover -eq 1 ] &&[[ ! -f "test_returned" ]]; then
+			#
+			# Check to see if we used spot, and we are to recover if the system goes away.
+			#
+			if [[ -f test_started ]]; then
+				sp_check=`grep spot_range: ansible_vars_main.yml`
+				if [[ $sp_check == *","* ]]; then
+					rm -rf tf
+					#
+					# Next attempt is with out spot pricing.
+					#
+					grep -v "spot_range:" ansible_vars_main.yml | grep -v spot_range: > spot_repair
```

Comment:
I'm curious why you're "grep -v"-ing twice here, the input through the pipe should be excluding spot_range: already.